subprojects/groovy-sql/src/test/groovy/groovy/sql/SqlBatchTest.groovy

```diff
@@ -119,6 +121,44 @@ class SqlBatchTest extends GroovyTestCase {
         // FINE: Successfully executed batch with 1 command(s)
     }
 
+    void testWithBatchHavingSizeSameSizeAsStatements() {
+        def numRows = sql.rows("SELECT * FROM PERSON").size()
+        assert numRows == 3
+        def myOthers = ['f4':'l4','f5':'l5','f6':'l6','f7':'l7']
+        def result = sql.withBatch(myOthers.size(), "insert into PERSON (id, firstname, lastname) values (?, ?, ?)") { ps ->
+            myOthers.eachWithIndex { k, v, index ->
+                def id = index + numRows + 1
+                ps.addBatch(id, k, v)
+            }
+        }
+        assert result == [1] * myOthers.size()
+        assert sql.rows("SELECT * FROM PERSON").size() == numRows + myOthers.size()
+        // end result the same as if no batching was in place but logging should show:
+        // FINE: Successfully executed batch with 4 command(s)
+    }
+
+    void testWithBatchNothingSpecified() {
+        def numRows = sql.rows("SELECT * FROM PERSON").size()
+        assert numRows == 3
+
+        def msg = GroovyAssert.shouldFail {
+            sql.withBatch(3, "insert into PERSON (id, firstname, lastname) values (?, ?, ?)") { ps ->
+                // Do nothing - not a good practice at all...
+            }
+        }
+        System.err.println(msg)
+    }
+
+    void testWithBatchNoPreparedStatementNothingSpecified() {
+        def numRows = sql.rows("SELECT * FROM PERSON").size()
+        assert numRows == 3
+
+        def result = sql.withBatch { ps ->
+            // Do nothing - not a good practice at all...
+        }
+        assert result == [] as int[]
```

Comment:
It is quite surprising at first sight that HSQL driver behaves differently if `executeBatch()` is called with nothing added to the batch, whether a preparedStatement is used or not.

I felt like it's not Groovy responsibility to "suppress" the exception for the preparedStatement case, and I wanted to test-specify those 2 behaviours as they are the current ones.